File: src/1310-XOR-Queries-of-a-Subarray/1310.cpp
class Solution 
{
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) 
    {
        vector<int> A = {0};
        for (int a : arr) A.emplace_back(A.back() ^ a);
        
        vector<int> res;
        for (auto& it : queries) res.emplace_back(A[it[0]] ^ A[it[1] + 1]);
        return res;
    }
};

File: src/1310-XOR-Queries-of-a-Subarray/1310.js
var xorQueries = function(arr, queries) {
    let A = new Array(arr.length + 1);
    A.fill(0);
    for (let i = 1; i <= arr.length; i++) {
        A[i] = A[i - 1] ^ arr[i - 1];
    } 

    let res = new Array(queries.length);
    for (let i = 0; i < queries.length; i++) {
        res[i] = A[queries[i][0]] ^ A[queries[i][1] + 1];
    }
    return res;
};